caikit/runtime/train.py

```diff
@@ -0,0 +1,198 @@
+
+    # Required Args
+    parser.add_argument(
+        "--training-kwargs",
+        "-k",
+        required=True,
+        help="Json string or json file pointer with keyword args for the training job",
+    )
+    parser.add_argument(
+        "--module",
+        "-m",
+        required=True,
+        help="Module name (package.Class) or UID to train",
+    )
+    parser.add_argument(
+        "--model-name",
+        "-n",
+        required=True,
+        help="Name to save the model under",
+    )
+
+    # Optional args
+    parser.add_argument(
+        "--save-path",
+        "-s",
+        default=".",
+        help="Path to save the output model to",
+    )
+    parser.add_argument(
+        "--library",
+        "-l",
+        nargs="*",
+        help="Libraries that need to be imported to register the module to train",
+    )
+    parser.add_argument(
+        "--trainer",
+        "-t",
+        default=None,
+        help="Trainer config name to use",
+    )
+    parser.add_argument(
+        "--save-with-id",
+        "-i",
+        action="store_true",
+        default=False,
+        help="Include the training ID in the save path",
+    )
+
+    args = parser.parse_args()
+    config_logging()
+
+    # Initialize top-level kwargs
+    train_kwargs = {
+        "save_path": args.save_path,
+        "save_with_id": args.save_with_id,
+        "model_name": args.model_name,
+    }
+    if args.trainer is not None:
+        train_kwargs["trainer"] = args.trainer
+
+    # Import libraries to register modules
+    for library in args.library or []:
+        log.info("<COR88091092I>", "Importing library %s", library)
+        importlib.import_module(library)
+
+    # Try to import the root library of the provided module. It's ok if this
+    # fails since the module may be a UID
+    mod_root_lib = args.module.split(".")[0]
+    try:
+        importlib.import_module(mod_root_lib)
+    except ImportError:
+        log.debug("Unable to import module root lib: %s", mod_root_lib)
+
+    # Figure out the module to train
+    mod_reg = module_registry()
+    mod_pkg_to_mod = {
+        f"{mod.__module__}.{mod.__name__}": mod for mod in mod_reg.values()
+    }
+    module: Type[ModuleBase] = mod_reg.get(args.module, mod_pkg_to_mod.get(args.module))
+    error.value_check(
+        "<COR03876205E>",
+        module is not None,
+        "Unable to find module {} to train",
+        args.module,
+    )
+
+    # Read training kwargs
+    try:
+        if os.path.isfile(args.training_kwargs):
+            with open(args.training_kwargs, encoding="utf-8") as handle:
+                training_kwargs = json.load(handle)
+        else:
+            training_kwargs = json.loads(args.training_kwargs)
+    except json.decoder.JSONDecodeError:
+        log.error(
+            "<COR65834760E>",
+            "--training-kwargs must be valid json or point to a valid json file",
+        )
+        raise
```

Comment:
not sure how the error handling works in the new system, but do we need to take above message and throw it back here with a `raise`?

Reply:
`raise` with no argument will automatically raise the top exception on the exception stack, so it will be the `jsoin.decoder.JSONDecodeError`. I figured it was best to just let that pass through for ease of debugging